subsys/bluetooth/audio/audio.c

```diff
@@ -18,6 +18,53 @@
 
 LOG_MODULE_REGISTER(bt_audio, CONFIG_BT_AUDIO_LOG_LEVEL);
 
+int bt_audio_data_parse(const uint8_t ltv[], size_t size,
+			bool (*func)(struct bt_data *data, void *user_data), void *user_data)
+{
+	CHECKIF(ltv == NULL) {
+		LOG_DBG("ltv is NULL");
+
+		return -EINVAL;
+	}
+
+	CHECKIF(func == NULL) {
+		LOG_DBG("func is NULL");
+
+		return -EINVAL;
+	}
+
+	for (size_t i = 0; i < size;) {
+		const uint8_t len = ltv[i++];
+		struct bt_data data;
+
+		if (i + len > size || len < sizeof(data.type)) {
+			LOG_DBG("Invalid len %u at i = %zu", len, i - 1);
+
+			return -EINVAL;
+		}
+
+		data.type = ltv[i++];
+		data.data_len = len - sizeof(data.type);
+
+		if (data.data_len > 0) {
+			data.data = &ltv[i];
+		} else {
+			data.data = NULL;
+		}
+
+		if (!func(&data, user_data)) {
+			return -ECANCELED;
+		}
+
+		/* Since we are incrementing i by the value_len, we don't need to increment it
+		 * further in the `for` statement
+		 */
+		i += data.data_len;
+	}
+
+	return 0;
+}
```

Comment:
```suggestion
void bt_audio_data_parse(uint8_t ltv[], size_t size,
			 bool (*func)(struct bt_data *data, void *user_data), void *user_data)
{
	struct net_buf_simple buf;

	net_buf_simple_init_with_data(&buf, ltv, size);

	bt_data_parse(&buf, func, user_data);
}
```

Reply:
Can't do this; the reason why I had to create a new function for this, is that `net_buf_simple` does not work with `const` data, which is required for LE Audio, as e.g. the codec capability data the stack provide to the application in e.g. the BAP Unicast Client is `const`.